packages/jsii-java-runtime-test/project/src/test/java/software/amazon/jsii/testing/ComplianceTest.java

```diff
@@ -968,6 +973,15 @@ public void receiveInstanceOfPrivateClass() {
         assertTrue(new ReturnsPrivateImplementationOfInterface().getPrivateImplementation().getSuccess());
     }
 
+    @Test
+    public void objRefsAreLabelledUsingWithTheMostCorrectType() {
+        final PublicClass classRef = Constructors.makeClass();
+        final IPublicInterface ifaceRef = Constructors.makeInterface();
+
+        assertTrue(classRef instanceof InbetweenClass);
```

Comment:
I don't see how this can be true. But if it works, then it works.

Reply:
This property being true is literally the whole point of this change.... 🤔